src/utils/github.ts

```diff
@@ -317,44 +318,46 @@ export async function getGitHubUserInfo(headers: HeadersInit): Promise<GithubUse
   const data = await response.json();
 
   const userInfo: GithubUserInfo = {
-    name: data.name,
-    login: data.login,
+    name: data.name || '',
+    login: data.login || '',
     email: ''
   };
 
   userInfo.email = await getGitHubUserPrimaryEmail(headers);
   return userInfo;
 }
 
-export async function getGitHubUsername(headers: HeadersInit): Promise<string> {
-  const response = await fetch(`${GITHUB_API_URL}/user`, {
+export async function getGitHubUserPrimaryEmail(headers: HeadersInit): Promise<string> {
+  const response = await fetch(`${GITHUB_API_URL}/user/public_emails`, {
     headers
   });
 
   if (!response.ok) {
     const errorText = await response.text();
-    console.error('Failed to fetch GitHub username:', response.status, errorText);
-    throw new Error('Failed to fetch GitHub username');
+    console.error('Failed to fetch GitHub email address:', response.status, errorText);
+    return '';
```

Comment:
@nerdalert DCO sign-off needs Name and public primary email address for DCO sign-off. Otherwise the DCO signoff will fail and user needs to go back and fix the github account to make it pass.

So i am wondering if we should pop-up a warning saying that user needs to setup the name and primary email address for their github account otherwise the DCO sign-off will fail. If press continue, we can set the email/name to empty, and if they press "Fix it", we can direct them to their github profile. Or we can even do the hard enforcement, where if the name and email is not set, we pop-up the warning and don't allow them to contribute knowledge and skill contribution. open for other thoughts as well.

Reply:
@nimbinatus ah got it. But i think this masked email need to be set as a primary email address, otherwise the DCO will fail. So to clarify, user need to have a name and primary email address (not necessary that it should be public or not masked).